### ISSUE TITLE
Add rendering for natural=strait

### ISSUE THREAD

**Issue by matthijsmelissen** [gravitystorm/openstreetmap-carto#804]
Straits are important landforms, some of them are very well known, e.g. Strait of Gibraltar or Strait of Magellan. Rendering them would be great. Right now, natural=strait is widely used, but often combined with place=locality, in order to force the rendering. place=locality could be avoided by rendering natural=strait directly.

See also https://trac.openstreetmap.org/ticket/4190.

**Comment by anderfo** [comment]
What happened to this? Looks like it's still not rendered even at high zoom - being tagged on node, polygon or multipolygon...but on the "Humanitarian" layer it is rendered.

BTW I asked on the wiki/talk a few months ago about why a strait cannot be tagged as a line, while a valley can....no response yet: https://wiki.openstreetmap.org/wiki/Talk:Tag:natural%3Dstrait
The geometry of a strait is rather similar to a valley.
I'd still prefer polygons (or even multipolygons, although not permitted by the wiki), for the reasons mentioned by gravitystorm above.

**Comment by imagico** [comment]
> The geometry of a strait is rather similar to a valley.

Actually no, a strait is a narrow between land masses, the best land analogy would be a saddle.  Valley like straits are usually either submerged glacial troughs (i.e. fjords open at both ends) - like the Strait of Magellan or submerged river shaped valleys - like the Bosporus.  The vast majority of straits are neither, they are usually just passages between islands or islands and continents.

Most of the big straits around the world are either mapped as nodes or as ways with 2-4 nodes where the location of the start and end node usually have no particular meaning, they essentially only specify direction:

http://www.openstreetmap.org/way/146655069
http://www.openstreetmap.org/way/146655079
http://www.openstreetmap.org/way/146655070

The problem here is not mapping, the problem is rendering in a way that does not give wrong incentives to mapping.  If you look for example here:

http://overpass-turbo.eu/s/dGz

If you'd use the way length (or in case of area mapping the area size) for importance rating mappers would enlarge or shrink the features used for mapping to get the desired outcome - without any basis in reality.  Actual importance of straits is best determined from their width and the size of islands or continents they separate.

**Comment by dieterdreist** [comment]
2016-01-12 12:17 GMT+01:00 Christoph Hormann <notifications@github.com>:

> Actual importance of straits is best determined from their width and the
> size of islands or continents they separate.




actual importance is connected to the availability of alternatives but much
more to the importance for trade / amount of traffic, e.g. the Bosporus is
important because it's the only connection between the Black Sea and the
Mediterranean Sea, the Strait of Hormuz is important because it's the only
connection between the Persian Gulf and the Ocean and because there's a lot
of oil to transport, if the ships were transporting sand or milk it would
be much less important ;-)

tl;dr;
importance is relying much more on different factors than what
geometrically can be determined.

**Comment by imagico** [comment]
I am talking about cartographic importance as a natural, physical geography feature which has little connection with importance as a shipping route.  The Fram Strait is a very important strait in terms of physical geography (probably second most important globally after the Drake Passage) but has zero significance as a shipping route.

**Comment by dieterdreist** [comment]
sent from a phone

> Am 12.01.2016 um 13:18 schrieb Christoph Hormann <notifications@github.com>:
> 
> The Fram Strait is a very important strait in terms of physical geography (probably second most important globally after the Drake Passage) but has zero significance as a shipping route.


also close to zero importance for this style ;-)
I'm interested in showing those straits that are important for a general map, like the strait of Gibraltar or the Bering Strait (no problem showing the Fram Strait, just that's not something I d consider prioritary)

**Comment by imagico** [comment]
Well - a world map showing the Bering Strait and Strait of Gibraltar but not the Fram Strait and Drake Passage would IMO be rather annoying and also counterproductive in terms of mapper feedback here.

**Comment by anderfo** [comment]
OK, maybe more similar to a saddle than a valley in some cases. But anyway it's a feature with some length and width which are both hard to define exactly. Cheating on length or area in order to trick the style sheet is also possible for a valley.

The idea of defining "strait importance" based on the width and size of islands or continents they separate is a possibility, although it's probably harder to implement than using a simple way length or area of the mapped feature.
For a strait in a lake, the importance would of course have to be calculated in a different way (e.g. based on the size of the two bays/parts of the lake it connects). But I don't know if you have a different name for straits in lakes than in the sea. In my language they are identical.

**Comment by imagico** [comment]
A strait always has a width, usually well defined as the minimum distance between the land masses it separates but a length only in a few cases.  You will have a hard time specifying a length for any of the straits shown in http://overpass-turbo.eu/s/dGz

My main point is that however straits are rendered in this style it should not be done in a way that requires mappers to map something [non-verifiable](http://wiki.openstreetmap.org/wiki/Verifiability).

For starters i think rendering straits as point labels on either nodes, ways or areas with a relatively high starting zoom level would already be an improvement.  As directed point labels based on way direction, node with direction tag and area medial axis would be better.  With coastline data in the database using the distance to the nearest coastline/lake outline to decide on the starting zoom level would be an option as well - although it would probably make more sense to do this as a preprocessing step for performance reasons.

**Comment by mboeringa** [comment]
> The problem with polygons here is that they would ultimately always be wrong and painful in maintenance. Practical mapping uses almost exclusively nodes and lines. **Imagining an archipelago with various straits and channels between the islands and all of them mapped as polygons makes me shudder **- and there would be no additional useful information in a polygon

I agree polygons are not the best solution for straits, although features like fjords have been entered as such as well, but those are mostly better defined. Personally, allowing and using lines (non closed ways) as a rough baseline for labelling in my opinion would be the best option, and I support this in my personal ArcGIS Renderer by using them (in a similar way as for valleys and fjords). Yes, I know some of you are very much opposed to allowing mappers control of labelling or label positioning, but for these type of non very clearly defined features, I think it is the only viable option in the long run. It is not like natural features like straits, deserts and even valleys have a clear "fence" that delineates their exact boundary, like a fenced of garden...

> The key to prevent people from abusing nodes to manipulate label placement is to not use the node directly but only as a hint **and infer the actual label location from the surrounding (in this case the coastline)**.

This type of logic is virtually impossible to implement automatic, what you're asking is a "cartographers human intelligence / consciousness". I really don't see a viable option for this, and certainly non that doesn't suffer from a plethora of ambiguous results or outcomes of far to complicated routines.

> If you'd use the way length (or in case of area mapping the area size) for importance rating **mappers would enlarge or shrink the features used for mapping to get the desired outcome** - without any basis in reality. Actual importance of straits is best determined from their width and the size of islands or continents they separate.

You are assuming the worst before its even proven to happen in reality. Most labelling results for points, lines and polygons I have seen for valleys and fjords in my personal ArcGIS Renderer, indicate users being fairly trustworthy and

Anyway, straits are non-delineable from the start, so how you would even judge the mappers were working *"without any basis in reality"*???

> My main point is that however straits are rendered in this style it should not be done in a way that requires mappers to map something non-verifiable.

See my remark immediately above, I don't agree there any way straits could be verifiable in the first place. There just a stretch of water between some land masses with no defined begin or end point. Any approximate location of a strait's name should therefore do, as long as it doesn't enter into territory of another well known named water body.

**Comment by kocio-pl** [comment]
> Any  approximate location of a strait's name should therefore do, as long as  it doesn't enter into (also often approximate and ill defined)  territory of another well known named water body.

I agree. The level of precision and verifiability is not a golden standard for all the elements and scales. It's also not true that nodes are always better than areas, because you will not get the same position when next person will try to tag it. This kind of verifiability belongs not to the position and shape, but only to some easy named elements, like `shop=convenience` - but even then you might not be sure (speaking from experience - maybe it's rather `shop=greengrocer` or `shop=alcohol`?). So please, let rely a bit on the mappers in a grassroot project like OSM.

I think we could render them in the same way as bays - when tagged as ways they are related to an area (like any other water area), but when tagged as points they still get rendered, but quite late (like z14+).

**Comment by DENelson83** [comment]
To render the name of a strait defined by an enclosed way or relation, why not just render the name once, at the single interior point which is the furthest away from any point on the boundary of the polygon?  I don't know how you would implement that kind of method, but I thought I would just drop such an idea here.

**Comment by kocio-pl** [comment]
Thanks for dropping it here. You have just described the new Mapnik algorithm for placing labels inside polygons, so this is what would happen once this code is created: https://github.com/mapnik/mapnik/pull/3780.

The numbers of uses are raising quite fast this year and there are more lines and relations than nodes:

https://taginfo.openstreetmap.org/tags/?key=natural&value=strait

![taghistory 46](https://user-images.githubusercontent.com/5439713/45602422-530dd200-ba1e-11e8-81eb-a4db0e0e085c.png)

**Comment by DENelson83** [comment]
And in fact, I have made relations for every named body of seawater, both bays and straits, on the coast of British Columbia, except for those that the Canada–US border cuts through.

**Comment by kocio-pl** [comment]
Great!

Is anybody willing to make the code?
